core/src/raw/adapters/kv/backend.rs

```diff
@@ -202,8 +202,11 @@ impl KvLister {
             } else {
                 EntryMode::FILE
             };
-
-            oio::Entry::new(&build_rel_path(&self.root, &v), Metadata::new(mode))
+            let mut path = build_rel_path(&self.root, &v);
+            if path.is_empty() {
```

Comment:
should we move this logic to `build_rel_path` ? it seems to appear almost everywhere.
which means, if `v` is root, we return `/`

Reply:
there is a similar problem with `build_abs_path`. if root is `/`, and path is `/`,  then it'll return `""`. so if we `create_dir("/")` in some backends, for example: `rocksdb`, it will create key `""`

Reply:
@Xuanwo what do you think about these two problem ?

Reply:
> should we move this logic to `build_rel_path` ? it seems to appear almost everywhere.

The purpose of `build_rel_path` is to calculate the relative path from `path` to `root`.

Returning an empty string indicates that it's the root itself. It's fine with me as long as nothing breaks.

> there is a similar problem with `build_abs_path`. if root is `/`, and path is `/`, then it'll return `""`.

This is intentional, as we have `build_abs_path` and `build_rooted_abs_path`.

Reply:
Maybe a helper function would be better ?
for `build_abs_path`, my concert is when setting the root as `/`, and writing ("/", "xxx"), we end up with a key-value pair  `"" -> "xxx"` in the backend. while it doesn't affect the users, it's still something to consider.
Overall, I think my major issue is that sometimes i get confused by the mixed use of `""` and `/`, and we need to convert them in many places

Reply:
Hi, as mentioned before, it's by design. Some storage services need the behavior that `build_abs_path` provides. For example, S3 should write the key as `abc` instead of `/abc`. It's a decision that the service implementer should make.

I hope this addresses your concern.

Reply:
Ok, got it